### DISCUSSION TITLE
What does ItemFromValueResolver (HxAutosuggest) Actualy Do?

### DISCUSSION THREAD

**Discussion by chobo2** [havit/Havit.Blazor#670] [Q&A]
Hi

I am looking at the example: https://havit.blazor.eu/components/HxAutosuggest

and I am unclear what this actually does

```
	private async Task<EmployeeDto> ResolveAutosuggestItemFromValue(int? value)
	{
		if (value is null)
		{
			return null;
		}

		return await DemoDataService.GetEmployeeByIdAsync(value.Value);
	}
```

Is it's whole purpose to set a default value in the textbox? I never see it run after initial load. 

Also once a list is loaded in say I type in the example "Smith" I see a few results pop up but once I click out of that box and go back in I do not see that populated list anymore.

Is it possible to see that list without typing in again? I see initial suggestions would sort of be what I would want, but I would want it based on what was typed in not just some random initial values.

**Comment by hakenr** [comment]
The `ItemFromValueResolver` is designed to convert a value into an item. This is crucial for the `HxAutosuggest` component, which is optimized for selecting values from a potentially large set of items. Unlike `HxSelect`, `HxAutosuggest` cannot be preloaded with all items due to its design. It uses `DataProvider` to fetch and display a limited set of items based on the user's input.

This feature becomes especially important when `HxAutosuggest` is used in a form to edit an existing object, like a `Project` with its `Project.ProjectManagerId`. In such cases, it's necessary to transform the initial `Value` (the `ProjectManagerId`) of the object into an `Item` (`EmployeeDto`) for display, as there isn't any user input to trigger `DataProvider` for item retrieval.

As for your second query, currently, `HxAutosuggest` does not retain the values it suggests. If you need to refine your search, you will have to re-enter the query.

**Reply by chobo2** [reply]
I am creating 3 boxes that will have this HxAutosuggest and at the end a user will hit the search button.  It  does not sound like in my case I will need "ItemFromValueResolver"?

Would I just have it return an empty object then as it seems required.
